analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -52,11 +52,31 @@ private ProgramState()
         public ProgramState SetOperationValue(IOperationWrapperSonar operation, SymbolicValue value) =>
             SetOperationValue(operation.Instance, value);
 
-        public ProgramState SetOperationValue(IOperation operation, SymbolicValue value) =>
-            this with { OperationValue = OperationValue.SetItem(operation, value) };
+        public ProgramState SetOperationValue(IOperation operation, SymbolicValue value)
+        {
+            if (value != null)
+            {
+                return this with { OperationValue = OperationValue.SetItem(operation, value) };
+            }
+            else if (OperationValue.ContainsKey(operation))
+            {
+                return this with { OperationValue = OperationValue.Remove(operation) };
+            }
+            return this;
+        }
```

Comment:
We can go for the simple version. 
```suggestion
        public ProgramState SetOperationValue(IOperation operation, SymbolicValue value) =>
            value is null
                ? return this with { OperationValue = OperationValue.Remove(operation) }
                : return this with { OperationValue = OperationValue.SetItem(operation, value) };
```

Reply:
Same below